exporter/prometheusexporter/accumulator_test.go

```diff
@@ -26,46 +26,173 @@ import (
 	"go.uber.org/zap"
 )
 
-func TestAccumulateDeltaAggregation(t *testing.T) {
-	tests := []struct {
-		name       string
-		fillMetric func(time.Time, pmetric.Metric)
-	}{
-		{
-			name: "Histogram",
-			fillMetric: func(ts time.Time, metric pmetric.Metric) {
-				metric.SetName("test_metric")
-				metric.SetEmptyHistogram().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
-				metric.SetDescription("test description")
-				dp := metric.Histogram().DataPoints().AppendEmpty()
-				dp.BucketCounts().FromRaw([]uint64{5, 2})
-				dp.SetCount(7)
-				dp.ExplicitBounds().FromRaw([]float64{3.5, 10.0})
-				dp.SetSum(42.42)
-				dp.Attributes().PutStr("label_1", "1")
-				dp.Attributes().PutStr("label_2", "2")
-				dp.SetTimestamp(pcommon.NewTimestampFromTime(ts))
-			},
-		},
+func TestAccumulateHistogram(t *testing.T) {
+	appendHistogram := func(startTs time.Time, ts time.Time, count uint64, sum float64, counts []uint64, bounds []float64, metrics pmetric.MetricSlice) {
+		metric := metrics.AppendEmpty()
+		metric.SetName("test_metric")
+		metric.SetEmptyHistogram().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+		metric.SetDescription("test description")
+		dp := metric.Histogram().DataPoints().AppendEmpty()
+		dp.ExplicitBounds().FromRaw(bounds)
+		dp.BucketCounts().FromRaw(counts)
+		dp.SetCount(count)
+		dp.SetSum(sum)
+		dp.Attributes().PutStr("label_1", "1")
+		dp.Attributes().PutStr("label_2", "2")
+		dp.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+		dp.SetStartTimestamp(pcommon.NewTimestampFromTime(startTs))
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			resourceMetrics := pmetric.NewResourceMetrics()
-			ilm := resourceMetrics.ScopeMetrics().AppendEmpty()
-			ilm.Scope().SetName("test")
-			tt.fillMetric(time.Now(), ilm.Metrics().AppendEmpty())
+	startTs1 := time.Now().Add(-6 * time.Second)
+	startTs2 := time.Now().Add(-5 * time.Second)
+	startTs3 := time.Now()
+	ts1 := time.Now().Add(-4 * time.Second)
+	ts2 := time.Now().Add(-3 * time.Second)
+	ts3 := time.Now().Add(-2 * time.Second)
+	ts4 := time.Now().Add(-1 * time.Second)
+	ts5 := time.Now().Add(1 * time.Second)
+
+	a := newAccumulator(zap.NewNop(), 1*time.Hour).(*lastValueAccumulator)
+
+	resourceMetrics1 := pmetric.NewResourceMetrics()
+	ilm1 := resourceMetrics1.ScopeMetrics().AppendEmpty()
+	ilm1.Scope().SetName("test")
+	// counts is one more than explicit bounds to account for the one implicit count/bucket for <=inf
+	appendHistogram(startTs2, ts3, 5, 2.5, []uint64{1, 3, 1, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm1.Metrics())
+	appendHistogram(startTs2, ts2, 4, 8.3, []uint64{1, 1, 2, 0, 0}, []float64{0.1, 0.5, 1, 10}, ilm1.Metrics())
+
+	m3 := ilm1.Metrics().At(0).Histogram().DataPoints().At(0)
+	m2 := ilm1.Metrics().At(1).Histogram().DataPoints().At(0)
```

Comment:
Why are these out of order and where is `m1`?

Reply:
The numbers were meant to be ordered chronologically I believe(I didn't write most of the tests). I re-wrote all of the tests just now to avoid these confusions; marking this as resolved